pkg/gather/clusterconfig/gather_logs.go

```diff
@@ -0,0 +1,105 @@
+package clusterconfig
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"k8s.io/klog/v2"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// gatherLogsFromPodsInNamespace collects logs from all pods in provided namespace
+//   - messagesToSearch are the messages to filter the logs(case-insensitive)
+//   - sinceSeconds sets the moment to fetch logs from(current time - sinceSeconds)
+//   - limitBytes sets the maximum amount of logs that can be fetched
+//   - logFileName sets the name of the file to save logs to.
+// Actual location is `config/pod/{namespace}/logs/{podName}/{fileName}.log`
+func gatherLogsFromPodsInNamespace(
+	g *Gatherer,
+	namespace string,
+	messagesToSearch []string,
+	sinceSeconds int64,
+	limitBytes int64,
+	logFileName string,
+) ([]record.Record, error) {
+	ctx := g.ctx
+
+	gatherKubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	coreClient := gatherKubeClient.CoreV1()
+
+	pods, err := coreClient.Pods(namespace).List(ctx, metav1.ListOptions{})
```

Comment:
This lists all the pods in the given namespace. It's unnecessary when you are interested only in "sdn-xxx" pods. You can use label selector - see https://github.com/openshift/insights-operator/pull/307/files#diff-89303e69a3ce0cbee54df62e174b25ca4c6476221800cc910e655b6e5d50dbfdR36. It would "app=sdn" in this case.